linera-core/src/client/validator_manager/cache.rs

```diff
@@ -0,0 +1,387 @@
+// Copyright (c) Zefchain Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::{collections::HashMap, sync::Arc};
+
+use linera_base::time::{Duration, Instant};
+
+#[cfg(with_metrics)]
+use super::manager::metrics;
+
+/// Cached result entry with timestamp for TTL expiration
+#[derive(Debug, Clone)]
+pub(super) struct CacheEntry<R> {
+    result: Arc<R>,
+    cached_at: Instant,
+}
+
+/// Cache for request results with TTL-based expiration and LRU eviction.
+///
+/// This cache supports:
+/// - Exact match lookups
+/// - Subsumption-based lookups (larger requests can satisfy smaller ones)
+/// - TTL-based expiration
+/// - Size-based LRU eviction
```

Comment:
It's not size-based, is it? Below we just use `cache.len() < self.max_cache_size`.

Reply:
Or just "LRU eviction"?